calibration/least_squares_calibration.py

```diff
@@ -0,0 +1,178 @@
+#%%
+import os
+import pandas as pd
+import numpy as np
+import matplotlib.pyplot as plt
+from rocketlogger.data import RocketLoggerData
+from sklearn import linear_model
+import yaml
+try:
+    from yaml import CLoader as Loader
+except ImportError:
+    from yaml import Loader
+
+#%%
+def load_data(cfg, datafiles):
+    df_list = []
+    for d in datafiles:
+        df = pd.read_csv(d)
+        df_list.append(df)
+    
+    data = pd.concat(df_list, ignore_index=True)
+    data = data.set_index("V")
+
+    data["I_in"] = data["I_in"] 
+    data["I_meas"] = data["V_i"] 
+    data["V_meas"] = data["V_2x"]
+    
+    return data
+
+#%%
+def load_rld(path, channel, start, stop, step):
+    rld = RocketLoggerData(path)
+    
+    source_range = np.arange(start, stop, step)
+    
+    def estimate_input(x):
+        diff = np.abs(source_range - x)
+        idx = np.argmin(diff)
+        return source_range[idx]
+
+    rld_df = pd.DataFrame(rld.get_data(), columns=rld.get_channel_names())
+    rld_df["source"] = rld_df[channel].apply(estimate_input)
+    rld_df["input"] = rld_df["source"]
+    rld_df = rld_df.set_index("source")
+    return rld_df
+
+#%%
+cfg_path = "data/config.yaml"
+datafiles = ["data/sps1_sweep_curr.csv"] # ':' not a valid file path in windows or mac
+evalfiles = ["data/sps3_temp_2022-10-15T18_37_56-07_00.csv"]
+
+#%%
+with open(cfg_path, "r") as f:
+    cfg = yaml.load(f, Loader=Loader)
+
+data = load_data(cfg, datafiles)
+
+v_input_cols = ["V_meas"]
+i_input_cols = ["I_meas"]
+
+v_model = linear_model.LinearRegression()
+v_model.fit(data[v_input_cols], data["V_in"])
+
+i_model = linear_model.LinearRegression()
+i_model.fit(data[i_input_cols], data["I_in"])
+
+eval_data = load_data(cfg, evalfiles)
+
+eval_data["V_fit"] = v_model.predict(eval_data[v_input_cols])
+eval_data["I_fit"] = i_model.predict(eval_data[i_input_cols])
+
+rld_v = load_rld("data/rl/20220826_200050_eval_v.rld", "V1", 0.1, 1, 0.05)
+rld_i = load_rld("data/rl/20220826_200050_eval_i.rld", "I1L", 45e-6, 454e-6, 22.7222e-6)
```

Comment:
The rdl files don't seem to match the range of the sps measurements.

Reply:
least_squares_calibration isn't used in my calibration